byway/src/main/webapp/utils.js

```diff
@@ -0,0 +1,33 @@
+/** Script that contains shared functions and variables */
+
+// map object used in the route page
+let map; 
+
+/**
+ * Uses placeId to retrieve details like coordinates, place name, etc.
+ * @param {String} placeId a textual identifier that uniquely identifies a place
+ * @return {Promise} result a Place object
+ */
+function findPlace(placeId) {
+  const request = {
+    placeId: placeId,
+    fields: ['name', 'geometry', 'place_id', 'photos', 'formatted_address']
+  }
+  const result = new Promise((resolve, reject) => {
+    placesService.getDetails(request, (result, status) => {
+      if(status == "OK") {
+        resolve(result);
+      } else {
+        alert("Status: " + status);
+        reject(new Error("Could not retrieve place result object from request."));
+      }
+    })
+  }).catch(error => {
+      alert(error);
+  });
```

Comment:
There's no point in rejecting if you're immediately going to catch your own error. Consider what this would look like if it was not using promises:

```
try {
  if (status == "OK") {
    return result; // resolve = return
  } else {
    alert("Status: " + status);
    throw new Error("..."); // reject = throw
  }
} catch (error) {
    alert(error);
}
```

Look at what happens in the error case:

1) `alert("Status: " + status);`
2) `alert(new Error("..."));` throw wrapped directly in catch + alert is basically equivalent to just sticking the error into the alert.
3) `return undefined;` Since there's no explicit return in the `catch` case, it is the same as returning `undefined`.